test/lib/payments.js

```diff
@@ -83,11 +125,69 @@ test('payments', async t => {
     })
     t.end()
   })
+
+  /* getAccountNeedsMarketplaceUpgrade */
+
+  t.test('getAccountNeedsMarketplaceUpgrade without billing', async t => {
+    const result = await getAccountNeedsMarketplaceUpgrade('000')
+    t.notOk(result)
+    t.end()
+  })
+
+  t.test('getAccountNeedsMarketplaceUpgrade with `free` plan', async t => {
+    const result = await getAccountNeedsMarketplaceUpgrade('123free')
+    t.notOk(result)
+    t.end()
+  })
+
+  t.test('getAccountNeedsMarketplaceUpgrade with stripe `personal` plan', async t => {
+    const result = await getAccountNeedsMarketplaceUpgrade('123')
+    t.notOk(result)
+    t.end()
+  })
+
+  t.test('getAccountNeedsMarketplaceUpgrade with stripe `org` plan', async t => {
+    const result = await getAccountNeedsMarketplaceUpgrade('123org')
+    t.notOk(result)
+    t.end()
+  })
+
+  t.test('getAccountNeedsMarketplaceUpgrade with `opensource` plan', async t => {
+    const result = await getAccountNeedsMarketplaceUpgrade('123opensource')
+    t.ok(result)
+    t.end()
+  })
+
+  t.test('getAccountNeedsMarketplaceUpgrade with `team` plan and under repo limit', async t => {
+    const result = await getAccountNeedsMarketplaceUpgrade('123team')
+    t.notOk(result)
+    t.end()
+  })
+
+  t.test('getAccountNeedsMarketplaceUpgrade with `team` plan and reached repo limit', async t => {
+    const { getAccountNeedsMarketplaceUpgrade } = proxyquire('../../lib/payments', { // Help! can't get this to be overwritten
+      './getCurrentlyPrivateAndEnabledRepos': async (accountId) => {
+        return 15
+      }
```

Comment:
@gr2m need your proxyrequire expertise!

Reply:
proxyquire can only mock other modules that are required in the module you are mocking. In the case above you are mocking a module which is loaded via `require('./getCurrentlyPrivateAndEnabledRepos') in `'../../lib/payments'`, but there is no such require, the getCurrentlyPrivateAndEnabledRepos method is defined in payments.js.

What you could do is sth like  this

```js
// on top of the file
const simple = require('simple-mock') // just an example, use any mock library you like
const payments = require('../../lib/payments')

// in the test
simple.mock(payments, 'getCurrentlyPrivateAndEnabledRepos').returnWith(15)
const result = await getAccountNeedsMarketplaceUpgrade('123team')
simple.restore()
```

Does that make sense?

Reply:
hm, it makes sense but it is not working either 😞